notebooks/Gradient Boosted Regression Trees - Single Target.ipynb

```diff
@@ -0,0 +1,646 @@
+   "metadata": {},
+   "source": [
+    "Gradient boosting regression in skl only supports single (1-d) targets.\n",
+    "Choose the single target you'd like to explore here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#output_columns = [\"Ca\",\"P\",\"pH\",\"SOC\",\"Sand\"]\n",
+    "output_columns=[\"Ca\"]\n",
+    "\n",
+    "X,targets=load.load_training_spectra()\n",
+    "\n",
+    "y=targets[output_columns]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Gradient Boosted Trees\n",
+    "Gradient boosted trees can be regarded as the smarter older sibling to the [Random Forest](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html#sklearn.ensemble.RandomForestRegressor) ensemble method.\n",
+    "Gradient Boosted Trees are iteratively corrected, with the residuals of the previous tree \n",
+    "providing the basis for building the subsequent tree.  \n",
+    "The user specifies the loss function to be optimized,\n",
+    "and here we use least squares regression,\n",
+    "the default loss function for scikit-learn's [GradientBoostingRegressor](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html).\n",
+    "More info and intuition about gradient boosted trees can be found [here](http://machinelearningmastery.com/gentle-introduction-gradient-boosting-algorithm-machine-learning/)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To help speed up the task of training these trees, \n",
+    "we perform some dimensionality reduction \n",
+    "and feature selection below"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "pca = skl.decomposition.PCA()\n",
+    "X_transformed = pca.fit_transform(X)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "X_train, X_test, y_train, y_test = skl.model_selection.train_test_split(X_transformed,y,test_size=0.2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Scikit-learn's **GradientBoostingRegressor** doesn't work for multiple targets, so we will train for one target at a time.  SKL does handle multioutput for instances like this with MultiOutputRegressor, but it seems to make info about the gradient boosting regressor less accessible (like feature importances).  For now we'll only look at one target, Ca."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "n_estimators=1000\n",
+    "clf_Ca=skl.ensemble.GradientBoostingRegressor(n_estimators=n_estimators)"
```

Comment:
I would avoid adding `_Ca` to variable names -- if someone re-runs this notebook but while trying to predict P, the variable names won't make sense!